File: frontend/src/types/integration.ts
export type GetnetEnvironment = "sandbox" | "production";

export interface GetnetEnvCredentials {
  seller_id: string | null;
  client_id: string | null;
  // No hay masking con sentido para un client_secret: sólo se informa si hay
  // uno guardado, el backend nunca lo devuelve.
  client_secret_set: boolean;
  last_verified_at: string | null;
  last_verified_ok: boolean | null;
  last_verified_message: string | null;
}

export interface GetnetIntegration {
  enabled: boolean;
  active_environment: GetnetEnvironment;
  sandbox: GetnetEnvCredentials;
  production: GetnetEnvCredentials;
  updated_at: string | null;
}

export interface GetnetEnvCredentialsInput {
  seller_id: string;
  client_id: string;
  // undefined = mantener el client_secret ya guardado de ESE ambiente.
  client_secret?: string;
}

export interface GetnetIntegrationInput {
  enabled: boolean;
  active_environment: GetnetEnvironment;
  sandbox: GetnetEnvCredentialsInput;
  production: GetnetEnvCredentialsInput;
}

export interface GetnetTestConnectionResult {
  ok: boolean;
  message: string;
}

export interface GetnetPublicConfig {
  enabled: boolean;
  environment: GetnetEnvironment;
  seller_id: string | null;
}

export interface IntegrationTestResult {
  ok: boolean;
  message: string;
}

export interface EmailIntegration {
  enabled: boolean;
  host: string | null;
  port: number;
  username: string | null;
  from_email: string | null;
  use_tls: boolean;
  // El backend nunca devuelve la contraseña: sólo informa si hay una guardada.
  password_set: boolean;
  last_verified_at: string | null;
  last_verified_ok: boolean | null;
  last_verified_message: string | null;
  updated_at: string | null;
}

export interface EmailIntegrationInput {
  enabled: boolean;
  host: string;
  port: number;
  username: string;
  from_email: string;
  use_tls: boolean;
  // undefined = mantener la contraseña ya guardada.
  password?: string;
}

File: backend/app/schemas/integration.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, EmailStr

Environment = Literal["sandbox", "production"]


class GetnetEnvCredentialsOut(BaseModel):
    seller_id: str | None = None
    client_id: str | None = None
    # No hay "últimos 4 dígitos" con sentido para un client_secret (no es una
    # tarjeta): sólo se informa si hay uno guardado, nunca se expone.
    client_secret_set: bool = False
    last_verified_at: datetime | None = None
    last_verified_ok: bool | None = None
    last_verified_message: str | None = None


class GetnetIntegrationOut(BaseModel):
    enabled: bool
    active_environment: Environment
    sandbox: GetnetEnvCredentialsOut
    production: GetnetEnvCredentialsOut
    updated_at: datetime | None = None


class GetnetEnvCredentialsUpdate(BaseModel):
    seller_id: str
    client_id: str
    # None = mantener el client_secret ya guardado para ESE ambiente (no se
    # reenvía en cada save; cada ambiente mantiene el suyo independiente).
    client_secret: str | None = None


class GetnetIntegrationUpdate(BaseModel):
    enabled: bool
    active_environment: Environment
    sandbox: GetnetEnvCredentialsUpdate
    production: GetnetEnvCredentialsUpdate


class GetnetTestConnectionResult(BaseModel):
    ok: bool
    message: str


class EmailIntegrationOut(BaseModel):
    """Estado de la integración SMTP para el panel. Nunca incluye la
    contraseña: sólo informa si hay una guardada (`password_set`)."""

    enabled: bool = False
    host: str | None = None
    port: int = 587
    username: str | None = None
    from_email: str | None = None
    use_tls: bool = True
    password_set: bool = False
    last_verified_at: datetime | None = None
    last_verified_ok: bool | None = None
    last_verified_message: str | None = None
    updated_at: datetime | None = None


class EmailIntegrationUpdate(BaseModel):
    enabled: bool
    host: str
    port: int = 587
    username: str = ""
    from_email: str
    use_tls: bool = True
    # None = mantener la contraseña ya guardada (no se reenvía en cada save).
    password: str | None = None


class EmailTestRequest(BaseModel):
    to: EmailStr


class IntegrationTestResult(BaseModel):
    ok: bool
    message: str


class GetnetPublicConfig(BaseModel):
    """Lo mínimo que necesita el checkout para cobrar con Getnet.

    Nunca incluye client_id/client_secret. Si la integración está apagada,
    seller_id tampoco se expone. Refleja siempre el ambiente ACTIVO
    (`active_environment`), nunca el otro guardado pero no en uso.
    """

    enabled: bool
    environment: Environment
    seller_id: str | None = None
